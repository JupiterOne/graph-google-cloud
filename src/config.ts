import {
  IntegrationExecutionContext,
  IntegrationLogger,
} from '@jupiterone/integration-sdk-core';
import { ServiceUsageClient } from './steps/service-usage/client';
import { IntegrationConfig, SerializedIntegrationConfig } from './types';
import { deserializeIntegrationConfig } from './utils/integrationConfig';
import { google } from 'googleapis';
import { handleApiClientError } from './google-cloud/client';
import { ServiceUsageListFilter } from './google-cloud/types';

export async function executeTestRequest(
  config: IntegrationConfig,
  logger: IntegrationLogger,
) {
  try {
    const client = google.serviceusage({ version: 'v1', retry: false });
    const googleClient = new ServiceUsageClient({ config }, logger);
    const auth = await googleClient.getAuthenticatedServiceClient();

    await client.services.list({
      parent: `projects/${config.serviceAccountKeyConfig.project_id}`,
      pageSize: 200,
      auth: auth,
      filter: ServiceUsageListFilter.ENABLED,
    });
  } catch (err) {
    throw handleApiClientError(err);
  }
}

export async function validateInvocation(
  context: IntegrationExecutionContext<SerializedIntegrationConfig>,
) {
  const { instance, logger } = context;
  const { config: serializedIntegrationConfig } = instance;
  const config = (context.instance.config = deserializeIntegrationConfig(
    serializedIntegrationConfig,
  ));

  await executeTestRequest(config, logger);
}
