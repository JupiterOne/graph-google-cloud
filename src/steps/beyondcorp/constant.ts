export const STEP_BEYONDCORP_APP_CONNECTOR = 'fetch-beyondcorp-app-connector';
export const BEYONDCORP_APP_CONNECTOR_CLASS = ['Network'];
export const BEYONDCORP_APP_CONNECTOR_TYPE =
  'google_cloud_beyondcorp_connector';

export const STEP_BEYONDCORP_GATEWAY = 'fetch-beyondcorp-gateway';
export const BEYONDCORP_GATEWAY_CLASS = ['Gateway'];
export const BEYONDCORP_GATEWAY_TYPE = 'google_cloud_beyondcorp_gateway';

export const STEP_BEYONDCORP_APPLICATION_ENDPOINT =
  'fetch-beyondcorp-application-endpoint';
export const BEYONDCORP_APPLICATION_ENPOINT_CLASS = ['NetworkEndpoint'];
export const BEYONDCORP_APPLICATION_ENDPOINT_TYPE =
  'google_cloud_beyondcorp_application_endpoint';

export const STEP_BEYONDCORP_APP_CONNECTION = 'fetch-beyondcorp-app-connection';
export const BEYONDCORP_APP_CONNECTION_CLASS = ['Network'];
export const BEYONDCORP_APP_CONNECTION_TYPE =
  'google_cloud_beyondcorp_connection';

export const STEP_BEYONDCORP_PARTNER_TENANT = 'fetch-beyondcorp-partner-tenant';
export const BEYONDCORP_ENTERPRISE_PARTNER_TENANT_TYPE =
  'google_cloud_beyondborp_partner_tenant';
export const BEYONDCORP_ENTERPRISE_PARTNER_TENANT_CLASS = ['Organization'];

export const STEP_BEYONDCORP_ENTERPRISE = 'fetch-beyondcorp-enterprise';
export const BEYONDCORP_ENTERPRISE_TYPE = 'google_cloud_beyondcorp_enterprise';
export const BEYONDCORP_ENTERPRISE_CLASS = ['Service'];

export const RELATIONSHIP_TYPE_APPLICATION_ENDPOINT_USES_GATEWAY =
  'google_cloud_beyondcorp_application_endpoint_uses_gateway';
export const STEP_APPLICATION_ENDPOINT_USES_GATEWAY_RELATIONSHIP =
  'fetch-application-endpoint-uses-gateway';

export const RELATIONSHIP_TYPE_APP_CONNECTION_HAS_APP_CONNECTOR =
  'google_cloud_beyondcorp_connection_has_connector';
export const STEP_APP_CONNECTION_HAS_APP_CONNECTOR_RELATIONSHIP =
  'fetch-app-connection-has-app-connector';

export const RELATIONSHIP_TYPE_APP_CONNECTION_HAS_APPLICATION_ENDPOINT =
  'google_cloud_beyondcorp_connection_has_application_endpoint';
export const STEP_APP_CONNECTION_HAS_APPLICATION_ENDPOINT_RELATIONSHIP =
  'fetch-app-connection-has-applifcation-endpoint';

export const RELATIONSHIP_TYPE_APP_CONNECTION_HAS_GATEWAY =
  'google_cloud_beyondcorp_connection_has_gateway';
export const STEP_APP_CONNECTION_HAS_GATEWAY_RELATIONSHIP =
  'fetch-app-connection-has-gateway';

export const RELATIONSHIP_TYPE_PROJECT_USES_APP_CONNECTOR =
  'google_cloud_project_uses_beyondcorp_connector';
export const STEP_PROJECT_USES_APP_CONNECTOR_RELATIONSHIP =
  'fetch-project-uses-app-connector';

export const RELATIONSHIP_TYPE_PROJECT_USES_APP_CONNECTION =
  'google_cloud_project_uses_beyondcorp_connection';
export const STEP_PROJECT_USES_APP_CONNECTION_RELATIONSHIP =
  'fetch-project-uses-app-connection';

export const RELATIONSHIP_TYPE_PROJECT_HAS_BEYONDCORP_ENTERPRISE =
  'google_cloud_project_has_beyondcorp_enterprise';
export const STEP_PROJECT_HAS_BEYONDCORP_ENTERPRISE_RELATIONSHIP =
  'fetch-project-has-beyondcorp-enterprise';

export const IngestionSources = {
  BEYONDCORP_CONNECTORS: 'beyondcorp_connectors',
  BEYONDCORP_GATEWAYS: 'beyondcorp_gateways',
  BEYONDCORP_CONNECTIONS: 'beyondcorp_connections',
  BEYONDCORP_APPLICATION_ENDPOINTS: 'beyondcorp_application_endpoints',
  BEYONDCORP_PARTNER_TENANT: 'beyondcorp_partner_tenant',
  BEYONDCORP_ENTERPRISE: 'beyondcorp_enterprise',
  APP_CONNECTION_HAS_APP_CONNECTOR_RELATIONSHIP:
    'app_connection_has_app_connector',
  APP_CONNECTION_HAS_APPLICATION_ENDPOINT_RELATIONSHIP:
    'app_connection_has_application_endpoint',
  APP_CONNECTION_HAS_GATEWAY_RELATIONSHIP: 'app_connection_has_gateway',
  PROJECT_USES_APP_CONNECTOR_RELATIONSHIP: 'project_uses_app_connector',
  PROJECT_USES_APP_CONNECTION_RELATIONSHIP: 'project_uses_app_connection',
  APPLICATION_ENDPOINT_USES_GATEWAY_RELATIONSHIP:
    'application_endpoint_uses_gateway',
  PROJECT_HAS_BEYONDCORP_ENTERPRISE: 'project_has_beyondcorp_enterprise',
};

export const MonitoringIngestionConfig = {
  [IngestionSources.BEYONDCORP_CONNECTORS]: {
    title: 'BeyondCorp App Connector',
    description: 'BeyondCorp App Connectors for GCP',
    defaultsToDisabled: false,
  },
  [IngestionSources.BEYONDCORP_GATEWAYS]: {
    title: 'BeyondCorp Gateways',
    description: 'BeyondCorp Gateways for GCP',
    defaultsToDisabled: false,
  },
  [IngestionSources.BEYONDCORP_ENTERPRISE]: {
    title: 'BeyondCorp Enterprise',
    description: 'BeyondCorp Enterprise for GCP',
    defaultsToDisabled: false,
  },
  [IngestionSources.BEYONDCORP_APPLICATION_ENDPOINTS]: {
    title: 'BeyondCorp Application endpoints',
    description: 'BeyondCorp Application endpoints for GCP',
    defaultsToDisabled: false,
  },
  [IngestionSources.BEYONDCORP_CONNECTIONS]: {
    title: 'BeyondCorp App Connections',
    description: 'BeyondCorp App Connection for GCP',
    defaultsToDisabled: false,
  },
  [IngestionSources.BEYONDCORP_PARTNER_TENANT]: {
    title: 'BeyondCorp Partner Tenant',
    description: 'BeyondCorp Partner Tenant for GCP',
    defaultsToDisabled: false,
  },
};
