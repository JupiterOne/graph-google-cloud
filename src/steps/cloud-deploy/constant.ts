import {
  RelationshipClass,
  RelationshipDirection,
  StepEntityMetadata,
  StepMappedRelationshipMetadata,
  StepRelationshipMetadata,
} from '@jupiterone/integration-sdk-core';
import { PROJECT_ENTITY_TYPE } from '../resource-manager';
import { StorageEntitiesSpec } from '../storage/constants';

// Steps
export const STEP_CLOUD_DEPLOY_DELIVERY_PIPELINE =
  'fetch-cloud-deploy-delivery-pipeline';
export const STEP_CLOUD_DEPLOY_AUTOMATION = 'fetch-cloud-deploy-automation';
export const STEP_CLOUD_DEPLOY_SERVICE = 'fetch-cloud-deploy-service';
export const STEP_PROJECT_HAS_CLOUD_DEPLOY_RELATIONSHIP =
  'build-project-has-cloud-deploy-relationship';
export const STEP_CLOUD_DEPLOY_SERVICE_HAS_DELIVERY_PIPELINE_RELAIONSHIP =
  'build-cloud-deploy-service-has-delivery-pipeline-relationship';
export const STEP_CLOUD_DEPLOY_AUTOMATION_TRIGGERS_DELIVERY_PIPELINE_AUTOMATION_RELATIONSHIP =
  'build-cloud-deploy-automation-triggers-delivery-pipeline-relationship';
export const STEP_CLOUD_DEPLOY_DELIVERY_PIPELINE_USES_GITHUB_REPO_RELAIONSHIP =
  'build-cloud-deploy-delivery-pipeline-uses-github-repo-relationship';
export const STEP_CLOUD_DEPLOY_DELIVERY_PIPELINE_USES_STORAGE_BUCKET_RELATIONSHIP =
  'build-cloud-deploy-delivery-pipeline-uses-storage-bucket-relationship';

// Entities
export const ENTITY_CLASS_CLOUD_DEPLOY_DELIVERY_PIPELINE = ['Workflow'];
export const ENTITY_TYPE_CLOUD_DEPLOY_DELIVERY_PIPELINE =
  'google_cloud_deploy_delivery_pipeline';

export const ENTITY_CLASS_CLOUD_DEPLOY_AUTOMATION = ['Rule'];
export const ENTITY_TYPE_CLOUD_DEPLOY_AUTOMATION =
  'google_cloud_deploy_automation';

export const ENTITY_CLASS_CLOUD_DEPLOY_SERVICE = ['Service'];
export const ENTITY_TYPE_CLOUD_DEPLOY_SERVICE = 'google_cloud_deploy_service';

// Relationships
export const RELATIONSHIP_TYPE_PROJECT_HAS_CLOUD_DEPLOY =
  'google_cloud_project_has_deploy_service';
export const RELATIONSHIP_TYPE_CLOUD_DEPLOY_SERVICE_HAS_DELIVERY_PIPELINE =
  'google_cloud_deploy_service_has_delivery_pipeline';
export const RELATIONSHIP_TYPE_CLOUD_DEPLOY_AUTOMATION_TRIGGERS_DELIVERY_PIPELINE =
  'google_cloud_deploy_automation_triggers_delivery_pipeline';
export const RELATIONSHIP_TYPE_CLOUD_DEPLOY_DELIVERY_PIPELINE_USES_STORAGE_BUCKET =
  'google_cloud_deploy_delivery_pipeline_uses_storage_bucket';

// Mapped Relationships
export const MAPPED_RELATIONSHIP_TYPE_CLOUD_DEPLOY_DELIVERY_PIPELINE_USES_GITHUB_REPO =
  'google_cloud_deploy_delivery_pipeline-uses-github-repo';

// IngestionSources
export const IngestionSources = {
  CLOUD_DEPLOY_DELIVERY_PIPELINES: 'cloud-deploy-delivery-pipelines',
  CLOUD_DEPLOY_AUTOMATION: 'cloud-deploy-automation',
  CLOUD_DEPLOY_SERVICE: 'cloud-deploy-service',
  PROJECT_HAS_CLOUD_DEPLOY_RELATIONSHIP:
    'project-has-cloud-deploy-relationship',
  CLOUD_DEPLOY_SERVICE_HAS_DELIVERY_PIPELINE_RELAIONSHIP:
    'cloud-deploy-service-has-delivery-pipeline-relationship',
  CLOUD_DEPLOY_AUTOMATION_TRIGGERS_DELIVERY_PIPELINE_AUTOMATION_RELATIONSHIP:
    'cloud-deploy-automation-triggers-delivery-pipeline-relationship',
  CLOUD_DEPLOY_DELIVERY_PIPELINE_USES_GITHUB_REPO_RELATIONSHIP:
    'cloud-deploy-delivery-pipeline-uses-github-repo-relationship',
  CLOUD_DEPLOY_DELIVERY_PIPELINE_USES_STORAGE_BUCKET_RELATIONSHIP:
    'cloud-deploy-delivery-pipeline-uses-storage-bucket-relationship',
};

// IngestionSources Configs
export const CodeDeployIngestionConfig = {
  [IngestionSources.CLOUD_DEPLOY_DELIVERY_PIPELINES]: {
    title: 'Google Cloud Deploy Delivery Pipelines',
    description: 'Cloud Deploy Delivery Pipelines.',
    defaultsToDisabled: false,
  },
  [IngestionSources.CLOUD_DEPLOY_AUTOMATION]: {
    title: 'Google Cloud Deploy Delivery Pipelines Automation',
    description: 'Cloud Deploy Delivery Pipelines Automation.',
    defaultsToDisabled: false,
  },
  [IngestionSources.CLOUD_DEPLOY_SERVICE]: {
    title: 'Google Cloud Deploy Service',
    description: 'Cloud Deploy Service.',
    defaultsToDisabled: false,
  },
  [IngestionSources.PROJECT_HAS_CLOUD_DEPLOY_RELATIONSHIP]: {
    title: 'Project Has Cloud Deploy Relationship',
    description: 'Build relationship between Project and Cloud Deploy Service.',
    defaultsToDisabled: false,
  },
  [IngestionSources.CLOUD_DEPLOY_SERVICE_HAS_DELIVERY_PIPELINE_RELAIONSHIP]: {
    title: 'Cloud Deploy Service Has Delivery Pipeline Relationship',
    description:
      'Build relationship between Delivery Pipeline and Cloud Deploy Service.',
    defaultsToDisabled: false,
  },
  [IngestionSources.CLOUD_DEPLOY_AUTOMATION_TRIGGERS_DELIVERY_PIPELINE_AUTOMATION_RELATIONSHIP]:
    {
      title: 'Cloud Deploy Automation Triggers Delivery Pipeline Relationship',
      description:
        'Build relationship between Automation and Delivery Pipeline.',
      defaultsToDisabled: false,
    },
  [IngestionSources.CLOUD_DEPLOY_DELIVERY_PIPELINE_USES_GITHUB_REPO_RELATIONSHIP]:
    {
      title: 'Cloud Deploy Delivery Pipeline Uses Github Repo Relationship',
      description:
        'Build relationship between Cloud Deploy Delivery Pipeline Uses Github Repo.',
      defaultsToDisabled: false,
    },
};

// IAM Permissions
export const CloudDeployPermissions = {
  STEP_CLOUD_DEPLOY_DELIVERY_PIPELINES: ['clouddeploy.deliveryPipelines.list'],
  STEP_CLOUD_DEPLOY_AUTOMATION: ['clouddeploy.automations.list'],
};

// Entities
export const Entities: Record<
  | 'CLOUD_DEPLOY_SERVICE'
  | 'CLOUD_DEPLOY_AUTOMATION'
  | 'CLOUD_DEPLOY_DELIVERY_PIPELINE',
  StepEntityMetadata
> = {
  CLOUD_DEPLOY_SERVICE: {
    resourceName: 'Cloud Deploy Service',
    _type: ENTITY_TYPE_CLOUD_DEPLOY_SERVICE,
    _class: ENTITY_CLASS_CLOUD_DEPLOY_SERVICE,
  },
  CLOUD_DEPLOY_AUTOMATION: {
    resourceName: 'Cloud Deploy Automation',
    _type: ENTITY_TYPE_CLOUD_DEPLOY_AUTOMATION,
    _class: ENTITY_CLASS_CLOUD_DEPLOY_AUTOMATION,
  },
  CLOUD_DEPLOY_DELIVERY_PIPELINE: {
    resourceName: 'Cloud Deploy Delivery Pipeline',
    _type: ENTITY_TYPE_CLOUD_DEPLOY_DELIVERY_PIPELINE,
    _class: ENTITY_CLASS_CLOUD_DEPLOY_DELIVERY_PIPELINE,
  },
};

// Relationships
export const Relationships: Record<
  | 'CLOUD_DEPLOY_AUTOMATION_TRIGGERS_DELIVERY_PIPELINE'
  | 'CLOUD_DEPLOY_DELIVERY_PIPELINE_USES_STORAGE_BUCKET'
  | 'PROJECT_HAS_CLOUD_DEPLOY'
  | 'CLOUD_DEPLOY_SERVICE_HAS_DELIVERY_PIPELINE',
  StepRelationshipMetadata
> = {
  CLOUD_DEPLOY_AUTOMATION_TRIGGERS_DELIVERY_PIPELINE: {
    _type: RELATIONSHIP_TYPE_CLOUD_DEPLOY_AUTOMATION_TRIGGERS_DELIVERY_PIPELINE,
    sourceType: ENTITY_TYPE_CLOUD_DEPLOY_AUTOMATION,
    _class: RelationshipClass.TRIGGERS,
    targetType: ENTITY_TYPE_CLOUD_DEPLOY_DELIVERY_PIPELINE,
  },
  CLOUD_DEPLOY_DELIVERY_PIPELINE_USES_STORAGE_BUCKET: {
    _type: RELATIONSHIP_TYPE_CLOUD_DEPLOY_DELIVERY_PIPELINE_USES_STORAGE_BUCKET,
    sourceType: ENTITY_TYPE_CLOUD_DEPLOY_DELIVERY_PIPELINE,
    _class: RelationshipClass.USES,
    targetType: StorageEntitiesSpec.STORAGE_BUCKET._type,
  },
  PROJECT_HAS_CLOUD_DEPLOY: {
    _type: RELATIONSHIP_TYPE_PROJECT_HAS_CLOUD_DEPLOY,
    sourceType: PROJECT_ENTITY_TYPE,
    _class: RelationshipClass.HAS,
    targetType: ENTITY_TYPE_CLOUD_DEPLOY_SERVICE,
  },
  CLOUD_DEPLOY_SERVICE_HAS_DELIVERY_PIPELINE: {
    _type: RELATIONSHIP_TYPE_CLOUD_DEPLOY_SERVICE_HAS_DELIVERY_PIPELINE,
    sourceType: ENTITY_TYPE_CLOUD_DEPLOY_SERVICE,
    _class: RelationshipClass.HAS,
    targetType: ENTITY_TYPE_CLOUD_DEPLOY_DELIVERY_PIPELINE,
  },
};

export const MappedRelationships: Record<
  'CLOUD_DEPLOY_DELIVERY_PIPELINE_USES_GITHUB_REPO',
  StepMappedRelationshipMetadata
> = {
  CLOUD_DEPLOY_DELIVERY_PIPELINE_USES_GITHUB_REPO: {
    _type:
      MAPPED_RELATIONSHIP_TYPE_CLOUD_DEPLOY_DELIVERY_PIPELINE_USES_GITHUB_REPO,
    sourceType: ENTITY_TYPE_CLOUD_DEPLOY_DELIVERY_PIPELINE,
    _class: RelationshipClass.USES,
    targetType: 'github_repo',
    direction: RelationshipDirection.FORWARD,
  },
};
