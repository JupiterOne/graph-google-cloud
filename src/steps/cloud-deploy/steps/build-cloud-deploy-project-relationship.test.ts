import {
  executeStepWithDependencies,
  Recording,
  StepTestConfig,
} from '@jupiterone/integration-sdk-testing';
import { invocationConfig } from '../../..';
import { integrationConfig } from '../../../../test/config';
import {
  getMatchRequestsBy,
  setupGoogleCloudRecording,
} from '../../../../test/recording';
import { STEP_PROJECT_HAS_CLOUD_DEPLOY_RELATIONSHIP } from '../constant';

describe(STEP_PROJECT_HAS_CLOUD_DEPLOY_RELATIONSHIP, () => {
  let recording: Recording;
  afterEach(async () => {
    if (recording) await recording.stop();
  });

  jest.setTimeout(450000);

  test.skip(STEP_PROJECT_HAS_CLOUD_DEPLOY_RELATIONSHIP, async () => {
    recording = setupGoogleCloudRecording({
      name: STEP_PROJECT_HAS_CLOUD_DEPLOY_RELATIONSHIP,
      directory: __dirname,
      options: {
        matchRequestsBy: getMatchRequestsBy(integrationConfig),
      },
    });

    const stepTestConfig: StepTestConfig = {
      stepId: STEP_PROJECT_HAS_CLOUD_DEPLOY_RELATIONSHIP,
      instanceConfig: integrationConfig,
      invocationConfig: invocationConfig as any,
    };

    const result = await executeStepWithDependencies(stepTestConfig);
    expect(result).toMatchStepMetadata(stepTestConfig);
  });
});
