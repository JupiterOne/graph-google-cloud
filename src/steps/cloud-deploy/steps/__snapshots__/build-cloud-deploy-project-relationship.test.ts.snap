// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Build Project and Cloud Deploy Service Relationship should collect data 1`] = `
{
  "collectedEntities": [
    {
      "_class": [
        "Service",
      ],
      "_key": "google_cloud_deploy_service:local-integration-instance",
      "_rawData": [],
      "_type": "google_cloud_deploy_service",
      "category": [
        "other",
      ],
      "createdOn": undefined,
      "displayName": "Google Cloud Deploy Service",
      "function": [
        "provisioning",
      ],
      "instanceId": "local-integration-instance",
      "name": "Google Cloud Deploy Service",
      "organizationId": "958457776463",
      "projectId": undefined,
    },
    {
      "_class": [
        "Account",
      ],
      "_key": "projects/167984947943",
      "_rawData": [
        {
          "name": "default",
          "rawData": {
            "createTime": "2021-05-31T15:15:56.125Z",
            "displayName": "j1-gc-integration-dev-v3",
            "etag": "W/"8f56735f6f3d890d"",
            "labels": {
              "firebase": "enabled",
            },
            "name": "projects/167984947943",
            "parent": "folders/306329820365",
            "projectId": "j1-gc-integration-dev-v3",
            "state": "ACTIVE",
            "updateTime": "2024-02-27T16:17:58.322524Z",
          },
        },
      ],
      "_type": "google_cloud_project",
      "createdOn": 1622474156125,
      "displayName": "j1-gc-integration-dev-v3",
      "id": "j1-gc-integration-dev-v3",
      "lifecycleState": "ACTIVE",
      "name": "projects/167984947943",
      "parent": "folders/306329820365",
      "projectId": "j1-gc-integration-dev-v3",
      "tag.firebase": "enabled",
      "updatedOn": 1709050678322,
      "webLink": "https://console.cloud.google.com/home/dashboard?project=j1-gc-integration-dev-v3",
    },
  ],
  "collectedRelationships": [
    {
      "_class": "HAS",
      "_fromEntityKey": "projects/167984947943",
      "_key": "projects/167984947943|has|google_cloud_deploy_service:local-integration-instance",
      "_toEntityKey": "google_cloud_deploy_service:local-integration-instance",
      "_type": "google_cloud_project_has_deploy_service",
      "displayName": "HAS",
    },
  ],
  "encounteredTypes": [
    "google_cloud_deploy_service",
    "google_cloud_project",
    "google_cloud_project_has_deploy_service",
  ],
  "numCollectedEntities": 2,
  "numCollectedRelationships": 1,
}
`;
